packages/react-components/src/components/InputBoxComponent.tsx

```diff
@@ -104,24 +167,315 @@ export const InputBoxComponent = (props: InputBoxComponentProps): JSX.Element =>
+      if (mentionTag !== undefined && mentionTag.plainTextBeginIndex !== undefined) {
+        if (selectionStartValue === null) {
+          updatedStartIndex = mentionTag.plainTextBeginIndex;
+          updatedEndIndex = mentionTag.plainTextEndIndex ?? mentionTag.plainTextBeginIndex;
+        } else {
+          const newSelectionIndex = findNewSelectionIndexForMention(
+            mentionTag,
+            inputTextValue,
+            event.currentTarget.selectionStart,
+            selectionStartValue
+          );
+
+          updatedStartIndex = newSelectionIndex;
+          updatedEndIndex = newSelectionIndex;
+        }
+      }
+    } else if (event.currentTarget.selectionStart !== event.currentTarget.selectionEnd) {
+      // Both e.currentTarget.selectionStart !== selectionStartValue and e.currentTarget.selectionEnd !== selectionEndValue can be true when a user selects a text by double click
+      if (event.currentTarget.selectionStart !== null && event.currentTarget.selectionStart !== selectionStartValue) {
+        // the selection start is changed
+        const mentionTag = findMentionTagForSelection(tagsValue, event.currentTarget.selectionStart);
+        if (mentionTag !== undefined && mentionTag.plainTextBeginIndex !== undefined) {
+          //TODO: here it takes -1 when it shouldn't, update selectionstart and end with mouse move and or touch move
+          updatedStartIndex = findNewSelectionIndexForMention(
+            mentionTag,
+            inputTextValue,
+            event.currentTarget.selectionStart,
+            selectionStartValue ?? -1
+          );
+        }
+      }
+      if (event.currentTarget.selectionEnd !== null && event.currentTarget.selectionEnd !== selectionEndValue) {
+        // the selection end is changed
+        const mentionTag = findMentionTagForSelection(tagsValue, event.currentTarget.selectionEnd);
+        if (mentionTag !== undefined && mentionTag.plainTextBeginIndex !== undefined) {
+          //TODO: here it takes -1 when it shouldn't, update selectionstart and end with mouse move and or touch move
+          updatedEndIndex = findNewSelectionIndexForMention(
+            mentionTag,
+            inputTextValue,
+            event.currentTarget.selectionEnd,
+            selectionEndValue ?? -1
+          );
+        }
+      }
+    }
+    // e.currentTarget.selectionDirection should be set to handle shift + arrow keys
+    if (event.currentTarget.selectionDirection === null) {
+      event.currentTarget.setSelectionRange(updatedStartIndex, updatedEndIndex);
+    } else {
+      event.currentTarget.setSelectionRange(updatedStartIndex, updatedEndIndex, event.currentTarget.selectionDirection);
+    }
+    setSelectionStartValue(updatedStartIndex);
+    setSelectionEndValue(updatedEndIndex);
+  };
+
+  const getInputFieldTextValue = (): string => {
+    /* @conditional-compile-remove(mention) */
+    return inputTextValue;
+    return textValue;
+  };
```

Comment:
why here is two `return ` in this function

Reply:
what does` @conditional-compile-remove(mention)` means here?

Reply:
The `conditional-compile-remove(mention)` removes the next token. In stable this means the second return is the only one there.